Core/src/Core/Filters/Algorithms/CalculateArrayHistogram.cpp

```diff
@@ -0,0 +1,315 @@
+#include "CalculateArrayHistogram.hpp"
+
+#include "Core/Filters/CalculateArrayHistogramFilter.hpp"
+
+#include "complex/Common/Range.hpp"
+#include "complex/DataStructure/DataArray.hpp"
+#include "complex/DataStructure/DataGroup.hpp"
+#include "complex/Utilities/ParallelDataAlgorithm.hpp"
+
+#include <chrono>
+
+using namespace complex;
+
+namespace
+{
+template <typename T>
+class GenerateHistogramFromData
+{
+public:
+  GenerateHistogramFromData(CalculateArrayHistogram& filter, const int32 numBins, const DataArray<T>& inputArray, Float64Array& histogram, std::atomic<usize>& overflow, float64 minRange = 0.0,
+                            float64 maxRange = 0.0)
+  : m_Filter(filter)
+  , m_NumBins(numBins)
+  , m_Min(minRange)
+  , m_Max(maxRange)
+  , m_InputArray(inputArray)
+  , m_Histogram(histogram)
+  , m_Overflow(overflow)
+  {
+  }
+  ~GenerateHistogramFromData() = default;
+
+  void generate(const usize start, const usize end) const
+  {
+    // tuple visulaization: Histogram = {(bin maximum, count), (bin maximum, count), ... }
+
+    float64 min = std::numeric_limits<float>::max();
+    float64 max = -1.0 * std::numeric_limits<float>::max();
+    if(m_Max == 0.0 && m_Min == 0.0)
+    {
+      const auto numOfElements = m_InputArray.getNumberOfComponents() * m_InputArray.getNumberOfTuples();
+      for(size_t i = 0; i < numOfElements; i++) // min and max in the input array
+      {
+        if(static_cast<float64>(m_InputArray[i]) > max)
+        {
+          max = static_cast<float64>(m_InputArray[i]);
+        }
+        if(static_cast<float64>(m_InputArray[i]) < min)
+        {
+          min = static_cast<float64>(m_InputArray[i]);
+        }
+      }
+      max = static_cast<float64>(max + 1); // ensure upper limit encapsulates max value
+      min = static_cast<float64>(min - 1); // ensure lower limit encapsulates min value
+    }
+    else
+    {
+      min = m_Min;
+      max = m_Max;
+    }
+
+    auto beginning = std::chrono::steady_clock::now();
+    const float64 increment = (max - min) / m_NumBins;
+    if(m_NumBins == 1) // if one bin, just set the first element to total number of points
+    {
+      m_Histogram[0] = max;
+      m_Histogram[1] = end;
+    }
+    else
+    {
+      for(usize i = start; i < end; i++)
+      {
+        m_Filter.updateThreadSafeProgress();
+        if(m_Filter.getCancel())
+        {
+          return;
+        }
+        const auto bin = std::floor((m_InputArray.at(i) - min) / increment);
+        if((bin >= 0) && (bin < m_NumBins))
+        {
+          m_Histogram[bin * 2 + 1]++;
+        }
+        else
+        {
+          m_Overflow++;
+        }
+      }
+    }
+
+    for(int64 i = 0; i < m_NumBins; i++)
+    {
+      m_Histogram[(i * 2)] = static_cast<float64>(min + (increment * (i + 1))); // load bin maximum into respective postion {(x, ), (x , ), ...}
```

Comment:
casting warnings here also. Try:
```
m_Histogram[(i * 2)] = static_cast<float64>(min + (increment * (i + 1.0F))); // load bin maximum into respective postion {(x, ), (x , ), ...}
```

The use of the "1.0F" will automatically do the conversion for you without the need for an explicit static_cast<>